dashboard/modules/metrics/dashboards/serve_dashboard_panels.py

```diff
@@ -51,42 +51,54 @@
+                legend="{{application, route}}",
+            ),
+            Target(
+                expr='sum(rate(ray_serve_num_grpc_requests{{application=~"$Application",application!~"",method=~"$gRPC_Method",{global_filters}}}[5m])) by (application, method)',
+                legend="{{application, method}}",
             ),
         ],
         grid_pos=GridPos(8, 0, 8, 8),
     ),
     Panel(
         id=8,
-        title="Error QPS per route",
-        description="Error QPS for each selected route.",
+        title="Error QPS per application",
+        description="Error QPS for each selected application.",
         unit="qps",
         targets=[
             Target(
-                expr='sum(rate(ray_serve_num_http_error_requests{{route=~"$Route",route!~"/-/.*",{global_filters}}}[5m])) by (route)',
-                legend="{{route}}",
+                expr='sum(rate(ray_serve_num_http_error_requests{{application=~"$Application",application!~"",route=~"$HTTP_Route",route!~"/-/.*",{global_filters}}}[5m])) by (application, route)',
+                legend="{{application, route}}",
+            ),
+            Target(
+                expr='sum(rate(ray_serve_num_grpc_error_requests{{application=~"$Application",application!~"",method=~"$gRPC_Method",{global_filters}}}[5m])) by (application, method)',
+                legend="{{application, method}}",
             ),
         ],
         grid_pos=GridPos(16, 0, 8, 8),
     ),
     Panel(
         id=12,
-        title="P50 latency per route",
-        description="P50 latency for selected routes.",
+        title="P50 latency per application",
+        description="P50 latency for selected applications.",
         unit="ms",
         targets=[
             Target(
-                expr='histogram_quantile(0.5, sum(rate(ray_serve_http_request_latency_ms_bucket{{route=~"$Route",route!~"/-/.*",{global_filters}}}[5m])) by (route, le))',
-                legend="{{route}}",
+                expr='histogram_quantile(0.5, sum(rate(ray_serve_http_request_latency_ms_bucket{{application=~"$Application",application!~"",route=~"$HTTP_Route",route!~"/-/.*",{global_filters}}}[5m])) by (application, route, le))',
+                legend="{{application, route}}",
+            ),
+            Target(
+                expr='histogram_quantile(0.5, sum(rate(ray_serve_grpc_request_latency_ms_bucket{{application=~"$Application",application!~"",method=~"$gRPC_Method",{global_filters}}}[5m])) by (application, method, le))',
+                legend="{{application, method}}",
             ),
             Target(
-                expr='histogram_quantile(0.5, sum(rate(ray_serve_http_request_latency_ms_bucket{{route=~"$Route",route!~"/-/.*",{global_filters}}}[5m])) by (le))',
+                expr='histogram_quantile(0.5, sum(rate({{__name__=~ "ray_serve_http_request_latency_ms_bucket|ray_serve_grpc_request_latency_ms_bucket",application=~"$Application",application!~"",{global_filters}}}[5m])) by (le))',
```

Comment:
You can make this more succinct:

```
__name__=~ "ray_serve_(http|grpc)_request_latency_ms_bucket"
```

Reply:
oh those are just regex, let me try it out, thanks for the suggestiong!